app/src/main/kotlin/dev/msfjarvis/aps/StoreRepository.kt

```diff
@@ -6,23 +6,75 @@
 
 package dev.msfjarvis.aps
 
-import android.util.Log
+import android.content.SharedPreferences
+import androidx.core.content.edit
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.github.ajalt.timberkt.Timber
 import dagger.Reusable
 import dev.msfjarvis.aps.db.dao.StoreDao
 import dev.msfjarvis.aps.db.entity.StoreEntity
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.withContext
+import dev.msfjarvis.aps.utils.PreferenceKeys
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.first
 import javax.inject.Inject
 
 @Reusable
-class StoreRepository @Inject constructor(private val storeDao: StoreDao) {
+class StoreRepository @Inject constructor(private val storeDao: StoreDao, private val prefs: SharedPreferences) {
 
-  suspend fun addStoreToDB(storeEntity: StoreEntity) {
+  private val _currentStore: MutableLiveData<StoreEntity> = MutableLiveData()
+  val currentStore: LiveData<StoreEntity> get() = _currentStore
+
+  private val job = Job()
+  private val scope = CoroutineScope(job + Dispatchers.Main)
+
+  init {
+    val storeId = getCurrentStore()
+    if (storeId != -1) {
+      scope.launch {
+        withContext(Dispatchers.IO) {
+          val store = storeDao.getStoreById(storeId).first()
+          _currentStore.postValue(store)
+        }
+      }
+    }
+  }
+
+  suspend fun addStoreToDB(storeEntity: StoreEntity, setCurrent: Boolean = false) {
     withContext(Dispatchers.IO) {
       storeDao.insertStore(storeEntity)
     }
     withContext(Dispatchers.Main) {
-      Log.d("StoreRepository", "Store added to DB")
+      Timber.d { "Store added to DB" }
```

Comment:
```suggestion
      d { "Store added to DB" }
```
the extension can be used directly